src/styles/itemTransition.scss

```diff
@@ -0,0 +1,50 @@
+.item {
+  &-enter {
+    max-height: 0;
+  }
+
+  &-enter-active {
+    overflow: hidden;
+    max-height: 58px;
+    transition: max-height 0.3s ease-in-out;
+  }
+
+  &-exit {
+    max-height: 58px;
+  }
+
+  &-exit-active {
+    overflow: hidden;
+    max-height: 0;
+    transition: max-height 0.3s ease-in-out;
+  }
+}
+
+.temp-item {
+  &-enter {
+    max-height: 58px;
+  }
+
+  &-enter-active {
+    overflow: hidden;
+    max-height: 58px;
+    transition: max-height 0.3s ease-in-out;
+  }
+
+  &-item-exit {
```

Comment:
The class name '.&-item-exit' seems incorrect. It should probably be '.&-exit' to be consistent with the rest of your code.